homeassistant/components/mawaqit/sensor.py

```diff
@@ -0,0 +1,203 @@
+"""Platform to retrieve Mawaqit prayer times information for Home Assistant."""
+
+import logging
+from typing import Any
+
+from homeassistant.components.sensor import SensorDeviceClass, SensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.storage import Store
+import homeassistant.util.dt as dt_util
+
+from . import utils
+from .const import (
+    DATA_UPDATED,
+    DOMAIN,
+    MAWAQIT_STORAGE_KEY,
+    MAWAQIT_STORAGE_VERSION,
+    PRAYER_TIMES_ICON,
+    SENSOR_TYPES,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Mawaqit prayer times sensor platform."""
+
+    client = hass.data[DOMAIN]
+    if not client:
+        _LOGGER.error("Error retrieving client object")
+
+    entities = []
+    for sensor_type in SENSOR_TYPES:
+        if sensor_type in [
+            "Fajr",
+            "Shurouq",
+            "Dhuhr",
+            "Asr",
+            "Maghrib",
+            "Isha",
+            "Jumua",
+            "Jumua 2",  # "Aid" and "Aid 2",
+            "Fajr Iqama",
+            "Shurouq Iqama",
+            "Dhuhr Iqama",
+            "Asr Iqama",
+            "Maghrib Iqama",
+            "Isha Iqama",
+            "Next Salat Name",
+            "Next Salat Time",
+            "Next Salat Preparation",
+        ]:
+            sensor = MawaqitPrayerTimeSensor(sensor_type, client)
+            entities.append(sensor)
+    async_add_entities(entities, True)
+
+    name = "My Mosque"
+    sensor1 = [MyMosqueSensor(name, hass)]
+    async_add_entities(sensor1, True)
+
+
+class MawaqitPrayerTimeSensor(SensorEntity):
+    """Representation of an Mawaqit prayer time sensor."""
+
+    def __init__(self, sensor_type, client) -> None:
+        """Initialize the Mawaqit prayer time sensor."""
+        self.sensor_type = sensor_type
+        self.client = client
+
+    @property
+    def name(self) -> str:
```

Comment:
For new integrations, we require the following:
- Support for [the new entity naming](https://developers.home-assistant.io/docs/core/entity#has_entity_name-true-mandatory-for-new-integrations)
- Services must be [translatable](https://developers.home-assistant.io/blog/2023/07/11/translating-services)
- Entities and their attributes should be [translatable](https://developers.home-assistant.io/blog/2023/03/27/entity_name_translations/)

Please adopt your code to add support for it.